### ISSUE TITLE
Modelsim isn't detected

### ISSUE THREAD

**Issue by pagdot** [michaelroland/fhlow#1]
Modelsim seem to work instead of Questasim, when pointing QuestaSim path to ModelSim, but it isn't detected automatic.

**Comment by michaelroland** [comment]
Currently, fhlow does not attempt to guess the simulation toolchain path (neither for QuestaSim nor for ModelSim). The default path for QuestaSim is hardcoded as `MY_QUESTASIM_PATH=C:\questasim64_10.4`. You are supposed to change the variable `MY_QUESTASIM_PATH` in your local `%USERPROFILE%\.fhlow\Paths.bat` file to point to your QuestaSim (or ModelSim) installation.

Feel free to provide a PR if you see an easy way to guess the path.

**Comment by pagdot** [comment]
I've found a way to add support. I wrote a Batch-Script to autodetect the location of Modelsim (tested with student edition) and Quartus using the registry.
I will try to also detect QuestaSim and will try to include it in the existing Path search if the predefined paths don't work.

```Batch
@echo off
setlocal enabledelayedexpansion

echo.ModelSim
echo.========
echo.

set MY_QUESTASIM_PATH=C:\questasim64_10.4

if not exist "%MY_QUESTASIM_PATH%" (
    echo Starting deep search...
    echo.
    for /F "skip=2 tokens=1,3" %%i in ('reg query HKEY_LOCAL_MACHINE\SOFTWARE /f ModelSim* /s /t REG_SZ /v ModelTech') do (
        if "%%i" == "ModelTech" (
            pushd "%%j\..\"
            set MY_QUESTASIM_PATH=!cd!
            popd
            set MY_QUESTASIM_EXEC_QUESTASIM=%%j\modelsim.exe
        )
    )    
)

echo Path: %MY_QUESTASIM_PATH%
echo Exec: %MY_QUESTASIM_EXEC_QUESTASIM%

echo.
echo.Quartus
echo.========
echo.

set MY_QUARTUS_PATH=

if not exist "%MY_QUARTUS_PATH%" (
    echo Starting deep search...
    echo.
    for /F "skip=2 tokens=1-3,5" %%i in ('reg query HKEY_LOCAL_MACHINE\SOFTWARE /f *Quartus* /s /t REG_SZ -v "Quartus Install Directory"') do (
        if "%%i %%j %%k" == "Quartus Install Directory" (
            set MY_QUARTUS_PATH=%%l
        )
    )    
)

echo Path: %MY_QUARTUS_PATH%

pause
```
Output on my PC:
```
ModelSim
========

Starting deep search...

Path: C:\Modeltech_pe_edu_10.4a
Exec: C:\Modeltech_pe_edu_10.4a\win32pe_edu\modelsim.exe

Quartus
========

Starting deep search...

Path: C:\intelFPGA_lite\17.1\quartus
Drücken Sie eine beliebige Taste . . .
```

**Comment by michaelroland** [comment]
Unfortunately, the registry-based approach does not seem to be usable with OEM versions of Modelsim. However, path discovery should be solved in version 2.2.
